bigtable/client/testing/inprocess_data_client.h

```diff
@@ -0,0 +1,65 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_TESTING_INPROCESS_DATA_CLIENT_H_
+#define GOOGLE_CLOUD_CPP_BIGTABLE_CLIENT_TESTING_INPROCESS_DATA_CLIENT_H_
+
+#include "bigtable/client/client_options.h"
+#include "bigtable/client/data_client.h"
+#include <google/bigtable/v2/bigtable.grpc.pb.h>
+
+namespace btproto = ::google::bigtable::v2;
+
+namespace bigtable {
+namespace testing {
+
+/**
+ * Connects to Cloud Bigtable's data manipulation APIs.
+ *
+ * This class is mainly for testing purpose, it enable use of a single embedded
+ * server
+ * for multiple test cases. This dataclient uses a pre-defined channel.
+ */
+class InProcessDtaClient : public bigtable::DataClient {
+ public:
+  InProcessDtaClient(std::string project, std::string instance,
+                     ClientOptions options,
+                     std::shared_ptr<grpc::Channel> channel)
+      : project_(std::move(project)),
+        instance_(std::move(instance)),
+        options_(std::move(options)),
```

Comment:
I think you can remove `options_`, and the `ClientOptions` parameter here too.

Reply:
removed